flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/StatusRecorder.java

```diff
@@ -140,6 +141,9 @@ public void updateStatusFromCache(CR resource) {
         } else {
             // Initialize cache with current status copy
             statusCache.put(key, objectMapper.convertValue(resource.getStatus(), ObjectNode.class));
+            if (ResourceLifecycleState.CREATED.equals(resource.getStatus().getLifecycleState())) {
+                statusUpdateListener.accept(resource, resource.getStatus());
```

Comment:
Are we going to emit duplicate events when the operator restarts? Should this occur when we actually create a deployment?

Reply:
I added the if condition to avoid sending status updates for those CRs the operator already reconciled. If the restart happens between detection of the CR and the observe phase, then we'll send a duplicate status with:
`The resource was created in Kubernetes but not yet handled by the operator`.  Which is still valid.